RIPS/rip-9999.md

```diff
@@ -0,0 +1,313 @@
+The `callData` parameter is interpreted as an RPL encoding of `N` call data byte arrays.
+The `gasLimits` parameter is interpreted as an RPL encoding of `N` 256-bit gas limit values.
+
+The field `atomicity` contains an RLP encoded array of `N` elements of `byte` "atomicity" values.
+Atomicity describes a behaviour in case of a top-level revert in one of the frames, and can take the following values:
+
+* `CONTINUE_ON_REVERT = 0` - ignores the reverted execution and continues executing the batch
+* `TERMINATE_ON_REVERT = 1` - stops the execution of the batch on the reverted execution without reverting previous calls in the batch. Does not execute the remainder 
+* `ROLLBACK_ON_REVERT = 2` - rolls back all state changes caused by all the execution frames in current transaction
+
+These parameters represent all the inputs necessary to perform `N` sub-transactions.
+
+
+### Gas-sponsored time-limited one-time delegated EOA transaction subtype
+
+```
+DELEGATED_EOA_SUBTYPE = x
+```
+
+The transaction payload for this subtype is extended with the following fields:
+```
+ destination
+ callGasLimit,
+ yParity, r, s
+```
+
+This transaction subtype extends the `SPONSORED_EOA_SUBTYPE` transactions with an ability to execute arbitrary
+contract code in the context of the EOA sender.
+
+Note that such transaction provides a one-time signature delegating execution to the `destination` contract,
+and this signature cannot be used repeatedly. The private key of the EOA will be needed to sign consequent delegations.
+
+### Gas-sponsored time-limited reusable authorization EOA transaction subtype
+
+```
+AUTHORIZED_EOA_SUBTYPE = x
+```
+
+The transaction payload for this subtype is extended with the following fields:
+```
+ chainId,
+ nonce,
+ invokerAddress,
+ callData,
+ callGasLimit
+ bigNonce, // TBD: in this transaction we would need both nonces - account state and 2D
+ validationGasLimit,
+ commit, // 32 bytes hash
+ yParity, r, s
+```
+
+This transaction subtype extends the `DELEGATED_EOA_SUBTYPE` transactions with an ability to authorize some
+contract code to act as a Smart Contract Account and control the context of the EOA sender.
+
+The transaction signature (`yParity`, `r`, `s`) are interpreted as an ECDSA signature on the `secp256k1` curve over
+the message `keccak256(MAGIC || chainId || nonce || implementationAddress || commit)`.
+
+The `commit` value is a hash values that the implementation will validate.
+It is passed to the validation frame as part of the encoded transaction structure. 
+
+Note that account's `nonce` is not incremented when `AUTHORIZED_EOA_SUBTYPE` transaction is executed so the same
+authorization can be used unlimited number of times. In order to withdraw authorization, the account must perform
+any action that will increment its nonce.
+
+Transaction hash uniqueness is therefore guaranteed by a `bigNonce` field for the `NonceManager` contract.
```

Comment:
The dependency on NonceManager in this subtype means that this RIP will require not just 7560 but also the one in [PR5](https://github.com/eth-infinitism/RIPs/pull/5/files).  This dependency only appears in the `AUTHORIZED_EOA_SUBTYPE` subtype, which may be a reason to move it to its own RIP.  The other EOA related subtypes only require 7560.